lib/parsers/exif_parser.rb

```diff
@@ -198,12 +198,12 @@ def load_sub_ifds(extended_io, sub_ifds_offsets)
     #    123 => { subIFD data...}
     #    456 => { another subIFD data...}
     # }
-    unless sub_ifds_offsets.empty?
-      EXIFR::TIFF::Data.open(extended_io) do |data|
+    return {} if sub_ifds_offsets.empty?
+
+    EXIFR::TIFF::Data.open(extended_io) do |data|
```

Comment:
Indentation is a bit weird here 😇  Also, you probably don't need `return` here either.